content/automate/ManageIQ/AutomationManagement/AnsibleTower/Operations/StateMachines/Job.class/__methods__/launch_ansible_job.rb

```diff
@@ -30,7 +30,7 @@ def main
 
                 def target
                   vm = @handle.root['vm'] || vm_from_request
-                  vm.name if vm
+                  vm.try(:floating_ip_addresses).try(:first) || vm.try(:ipaddresses).try(:first) if vm
```

Comment:
We also want to keep the `raise` logic from the other code sample I pointed.  If we do not have a VM for some reason, or the IP address is empty, this code will call Ansible without a limit which would try to run against all the inventory defined on the target JobTemplate.

https://github.com/ManageIQ/manageiq-content/blob/master/content/automate/ManageIQ/System/Request.class/__methods__/order_ansible_playbook.rb#L45-L52

Reply:
@tinaafitz 
The vm.name was used for running job templates on freshly MiQ provisioned VMs and the only way we can have the tower update its inventory and run the job is with the name and not the IP address. Since this is a method I think we should allow users to control if we should use the name or the IP address.